### PR TITLE
Add the "Two Images Showcase" block pattern

### PR DESCRIPTION
**Pull request by kjellr**
Addresses the first checkbox for #50. 

As part of implementation, this adds a "Image Frame" block style. It also adds the images to the  `assets/images/` directory, along with license information in `style.css`. 

## Screenshots

Screen Shot 2020-09-24 at 11 29 14 AM

Screen Shot 2020-09-24 at 11 09 46 AM

### CODE REVIEW

**Review by melchoyce — APPROVED** [review]
Looks great! Thanks, @kjellr.

File: inc/block-patterns.php [hunk 1]
```diff
@@ -12,33 +12,28 @@
...
 	register_block_pattern(
-		'',
+		'twentytwentyone/two-images-showcase',
 		array(
-			'title'      => __( '', 'twentytwentyone' ),
+			'title'      => __( 'Two Images Showcase', 'twentytwentyone' ),
 			'categories' => array( 'twentytwentyone' ),
-			'content'    => '',
+			'viewportWidth' => 1440,
+			'content'    => '<!-- wp:media-text {"mediaId":1747,"mediaLink":"' . get_template_directory_uri() . '/assets/images/la_mousme.jpg","mediaType":"image"} --><div class="wp-block-media-text alignwide is-stacked-on-mobile"><figure class="wp-block-media-text__media"><img src="' . get_template_directory_uri() . '/assets/images/la_mousme.jpg" alt="" size-full"/></figure><div class="wp-block-media-text__content"><!-- wp:image {"align":"center","width":400,"height":512,"sizeSlug":"large","className":"is-style-twentytwentyone-image-frame"} --><figure class="wp-block-image aligncenter size-large is-resized is-style-twentytwentyone-image-frame"><img src="' . get_template_directory_uri() . '/assets/images/the_berceuse_woman-rocking_a_cradle.jpg" alt="" width="400" height="512"/></figure><!-- /wp:image --></div></div><!-- /wp:media-text -->',
```
Comment (carolinan) [inline inc/block-patterns.php]: `get_template_directory_uri() ` needs to be escaped.

I am not sure what do about the alt attributes here.

File: assets/sass/01-settings/file-header.scss [hunk 2]
```diff
@@ -15,4 +15,12 @@ Tags: one-column, flexible-header, accessibility-ready, custom-colors, custom-me
 
 Twenty Twenty One WordPress Theme, (C) 2020 WordPress.org
 Twenty Twenty One is distributed under the terms of the GNU GPL.
+
+Twenty Twenty One bundles the following third-party resources:
```
Comment (carolinan) [inline assets/sass/01-settings/file-header.scss]: I would leave this only in the readme.txt, or the CSS file is going to grow much bigger.

File: inc/block-styles.php [hunk 3]
```diff
@@ -17,11 +17,10 @@
 	 */
 	function twenty_twenty_one_register_block_styles() {
 		register_block_style(
-			'core/latest-posts',
+			'core/image',
```
Comment (carolinan) [inline inc/block-styles.php:20]: +1 I think this style will be useful

File: assets/sass/01-settings/file-header.scss (hunk 2 above)
Comment (kjellr) [reply assets/sass/01-settings/file-header.scss]: Oh cool — I thought the license info needed to be in the stylesheet. I'll move to the readme. 👍

File: inc/block-patterns.php (hunk 1 above)
Comment (kjellr) [reply inc/block-patterns.php]: Yeah, I'm not sure about the alt attributes too, which is why I left them blank. 😕

File: assets/sass/01-settings/file-header.scss (hunk 2 above)
Comment (kjellr) [reply assets/sass/01-settings/file-header.scss]: Moved over in 72add4d

File: inc/block-patterns.php (hunk 1 above)
Comment (kjellr) [reply inc/block-patterns.php]: Escaped in 9df2990.

For the alt text, I added in the title, artist, and date for the paintings in 96f3c39.

**Comment by kjellr** [comment]
A couple of the PHPCS checks appear to be incorrect or out of date: 

```
FILE: ...nner/work/twentytwentyone/twentytwentyone/inc/block-patterns.php
10
----------------------------------------------------------------------
11
FOUND 2 ERRORS AFFECTING 2 LINES
12
----------------------------------------------------------------------
13
 18 | ERROR | Registering and deregistering editor blocks should be
14
    |       | done in a plugin, not in a theme. Found
15
    |       | register_block_pattern_category().
16
    |       | (WPThemeReview.PluginTerritory.ForbiddenFunctions.editor_blocks_register_block_pattern_category)
17
 29 | ERROR | Registering and deregistering editor blocks should be
18
    |       | done in a plugin, not in a theme. Found
19
    |       | register_block_pattern().
20
    |       | (WPThemeReview.PluginTerritory.ForbiddenFunctions.editor_blocks_register_block_pattern)
21
----------------------------------------------------------------------
```

These aren't custom blocks, they're just patterns. 😕

**Comment by carolinan** [comment]
That should have been fixed by https://github.com/WordPress/twentytwentyone/commit/4f1dff4c1762886162c48e9d9c3301574749160b 
But I see now it is only for register block style not register block pattern.

File: assets/sass/01-settings/file-header.scss (hunk 2 above)
Comment (carolinan) [reply assets/sass/01-settings/file-header.scss]: If you scroll down the readme there is already a section for assets:
https://github.com/WordPress/twentytwentyone/blob/trunk/readme.txt#L62

File: inc/block-patterns.php [hunk 4]
```diff
@@ -12,33 +12,28 @@
...
-			'content'    => '',
+			'title'         => __( 'Two Images Showcase', 'twentytwentyone' ),
+			'categories'    => array( 'twentytwentyone' ),
+			'viewportWidth' => 1440,
+			'content'       => '<!-- wp:media-text {"mediaId":1747,"mediaLink":"' . esc_url( get_template_directory_uri() ) . '/assets/images/la_mousme.jpg","mediaType":"image"} --><div class="wp-block-media-text alignwide is-stacked-on-mobile"><figure class="wp-block-media-text__media"><img src="' . esc_url( get_template_directory_uri() ) . '/assets/images/la_mousme.jpg" alt="&quot;La Mousmé&quot; by Vincent Van Gogh (1888)" size-full"/></figure><div class="wp-block-media-text__content"><!-- wp:image {"align":"center","width":400,"height":512,"sizeSlug":"large","className":"is-style-twentytwentyone-image-frame"} --><figure class="wp-block-image aligncenter size-large is-resized is-style-twentytwentyone-image-frame"><img src="' . esc_url( get_template_directory_uri() ) . '/assets/images/the_berceuse_woman-rocking_a_cradle.jpg" alt="&quot;The Berceuse, Woman Rocking a Cradle&quot; by Vincent Van Gogh (1889)" width="400" height="512"/></figure><!-- /wp:image --></div></div><!-- /wp:media-text -->',
```
Comment (carolinan) [inline inc/block-patterns.php]: All text strings in the theme needs to be translation ready.

**Review by jffng — COMMENTED** [review]
Looks great, this pattern is super nice 😍

I added a few commits to try and address the failing checks. The remaining failure is unrelated to this PR so once @carolinan's feedback has been addressed this works for me.

**Comment by aristath** [comment]
Pulled latest commits from `trunk`, all failing tests now pass.

File: assets/sass/01-settings/file-header.scss (hunk 2 above)
Comment (kjellr) [reply assets/sass/01-settings/file-header.scss]: Thanks, I missed that. Updated in c47e22a

File: inc/block-patterns.php (hunk 4 above)
Comment (kjellr) [reply inc/block-patterns.php]: Should be fixed in 140717d (and https://github.com/WordPress/twentytwentyone/pull/140/commits/14dbf068a8fb927c8ad9d5880e624d2d2aea482d)

File: inc/block-patterns.php [hunk 5]
```diff
@@ -12,33 +12,28 @@
...
+			'categories'    => array( 'twentytwentyone' ),
+			'viewportWidth' => 1440,
+			'content'       => '<!-- wp:media-text {"mediaId":1747,"mediaLink":"' . esc_url( get_template_directory_uri() ) . '/assets/images/la_mousme.jpg","mediaType":"image"} --><div class="wp-block-media-text alignwide is-stacked-on-mobile"><figure class="wp-block-media-text__media"><img src="' . esc_url( get_template_directory_uri() ) . '/assets/images/la_mousme.jpg" alt="' . __( '&quot;La Mousmé&quot; by Vincent Van Gogh (1888)', 'twentytwentyone' ) . '" size-full"/></figure><div class="wp-block-media-text__content"><!-- wp:image {"align":"center","width":400,"height":512,"sizeSlug":"large","className":"is-style-twentytwentyone-image-frame"} --><figure class="wp-block-image aligncenter size-large is-resized is-style-twentytwentyone-image-frame"><img src="' . esc_url( get_template_directory_uri() ) . '/assets/images/the_berceuse_woman-rocking_a_cradle.jpg" alt="' . __( '&quot;The Berceuse, Woman Rocking a Cradle&quot; by Vincent Van Gogh (1889)', 'twentytwentyone' ) . '" width="400" height="512"/></figure><!-- /wp:image --></div></div><!-- /wp:media-text -->',
```
Comment (carolinan) [inline inc/block-patterns.php]:
```suggestion
			'content'       => '<!-- wp:media-text {"mediaId":1747,"mediaLink":"' . esc_url( get_template_directory_uri() ) . '/assets/images/la_mousme.jpg","mediaType":"image"} --><div class="wp-block-media-text alignwide is-stacked-on-mobile"><figure class="wp-block-media-text__media"><img src="' . esc_url( get_template_directory_uri() ) . '/assets/images/la_mousme.jpg" alt="' . esc_attr__( '&quot;La Mousmé&quot; by Vincent Van Gogh (1888)', 'twentytwentyone' ) . '" size-full"/></figure><div class="wp-block-media-text__content"><!-- wp:image {"align":"center","width":400,"height":512,"sizeSlug":"large","className":"is-style-twentytwentyone-image-frame"} --><figure class="wp-block-image aligncenter size-large is-resized is-style-twentytwentyone-image-frame"><img src="' . esc_url( get_template_directory_uri() ) . '/assets/images/the_berceuse_woman-rocking_a_cradle.jpg" alt="' . esc_attr__( '&quot;The Berceuse, Woman Rocking a Cradle&quot; by Vincent Van Gogh (1889)', 'twentytwentyone' ) . '" width="400" height="512"/></figure><!-- /wp:image --></div></div><!-- /wp:media-text -->',
```

Comment (carolinan) [reply inc/block-patterns.php]: added esc_attr for the text in the alt attribute

Comment (kjellr) [reply inc/block-patterns.php]: Great catch, thank you!

**Comment by kjellr** [comment]
I _think_ this is good to go now, would someone mind giving it one last look? I can get going on some of the other patterns after this has merged.

**Comment by carolinan** [comment]
> I _think_ this is good to go now, would someone mind giving it one last look? I can get going on some of the other patterns after this has merged.

Looks good to me, moved things around, added some texts, tested with images with captions, and now with linked images.

Linked images needs focus and hover styles but they all do, not only the ones with the pattern or style.

**Comment by kjellr** [comment]
Awesome, thank you everyone!

**Pull request merged**